daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/Runner.scala

```diff
@@ -527,8 +534,16 @@ private[lf] class Runner(
       esf: ExecutionSequencerFactory,
       mat: Materializer,
   ): (Speedy.PureMachine, Future[SValue], Option[Runner.IdeLedgerContext]) = {
-    // use script.scriptIds to determine which runner
-
-    new v1.Runner(this).runWithClients(initialClients, traceLog, warningLog, canceled)
+    val damlScriptName = getPackageName(script.scriptIds.scriptPackageId)
+
+    damlScriptName.getOrElse(
+      throw new IllegalArgumentException("Couldn't get daml script package name")
+    ) match {
+      case "daml-script" =>
+        new v1.Runner(this).runWithClients(initialClients, traceLog, warningLog, canceled)
+      case "daml-script2" =>
+        new v2.Runner(this).runWithClients(initialClients, traceLog, warningLog, canceled)
+      case _ => throw new IllegalArgumentException("Invalid daml script package name")
```

Comment:
Could we have a more informative error saying what the attempted name was and what the two valid names are?

Reply:
Yep - I'll update this